server/cluster/cluster.go

```diff
@@ -2733,3 +2733,25 @@ func (c *RaftCluster) GetPausedSchedulerDelayAt(name string) (int64, error) {
 func (c *RaftCluster) GetPausedSchedulerDelayUntil(name string) (int64, error) {
 	return c.coordinator.getPausedSchedulerDelayUntil(name)
 }
+
+var (
+	onlineUnsafeRecoveryStatus = schedulingAllowanceStatusGauge.WithLabelValues("online-unsafe-recovery")
+	haltSchedulingStatus       = schedulingAllowanceStatusGauge.WithLabelValues("halt-scheduling")
+)
+
+// checkSchedulingAllowance checks if the cluster allows scheduling.
+func (c *RaftCluster) checkSchedulingAllowance() (bool, error) {
+	// If the cluster is in the process of online unsafe recovery, it should not allow scheduling.
+	if c.GetUnsafeRecoveryController().IsRunning() {
+		onlineUnsafeRecoveryStatus.Set(1)
+		return false, errs.ErrUnsafeRecoveryIsRunning.FastGenByArgs()
+	}
+	onlineUnsafeRecoveryStatus.Set(0)
+	// If the halt-scheduling is set, it should not allow scheduling.
+	if c.opt.IsSchedulingHalted() {
```

Comment:
It sounds that once the scheduling is halted, the scheduler won't accept or initiate the new scheduling task? What will happen to the accepted/running scheduling task? Do you need to drain the running scheduling tasks or pause the long running scheduling tasks before returning false below?

Reply:
Based on the current usage scenarios that require termination of scheduling, there is no such demand. We only need to ensure that no new scheduling is generated.